### ISSUE TITLE
Decision Proposal 012 - Payload Naming Conventions And Structures

### ISSUE THREAD

**Issue by JamesMBligh** [ConsumerDataStandardsAustralia/open-banking#12]

**Comment by bazzat** [comment]
The consensus of the ABA Online Banking Technical Working Group is that we support the recommendation.

**Comment by bazzat** [comment]
One WG member did suggest the addition of a conditional keyword similar to the UK spec at the field level. There are likely to be instances where fields would be conditional, especially when resources are heterogeneous (e.g. credit card account data vs savings account data).

**Comment by deboraelkin2** [comment]
I agree with the principles outlined. My only question is why is the links object mandatory for a successful response? Are there any other uses being contemplated besides pagination or including the self link? If it's just for pagination, there may be responses that are self contained and don't require any pagination info.
Also wondering the reason to include the self link (even though it's easy to include it does add up to response payload size) for non-GET operations?

**Comment by TKCOBA** [comment]
COBA supports Data61’s proposed approach, particularly the suggestion to leverage commonly used standards under the UK Open Banking system.

**Comment by JamesMBligh** [comment]
Some comments on the feedback to date before I close feedback down.

The self links, as well as the meta object are part of both the UK standards and the JSONAPI.org recommendations.  I would look to JSONAPI.org for explanations of the links section and its value.

With regards to fields in the meta object, the fact that the meta object is separate to the data object does not mean that it is anymore open to arbitrary field inclusion.  Each end point should specify if a field is present in the meta object and what conditions it should be present.  The pagination decision proposal gives an example where fields in the meta object are present to support paging.

I will add the concept of conditionality with the mandatory status of the fields.

-JB-

**Comment by JamesMBligh** [comment]
I have now closed consultation on this decision. A recommendation incorporating feedback will be made to the Chair in due course.

-JB-
